File: bindings/go/examples/transactions_with_function.go
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"fmt"
	"log"

	sdk "bindings/iota_sdk_ffi"
)

func isNilError(err error) bool {
	if sdkErr, ok := err.(*sdk.SdkFfiError); ok {
		return sdkErr == nil
	}
	return false
}

func main() {
	client := sdk.GraphQlClientNewDevnet()

	function := "0x3::iota_system::request_add_stake"
	transactions, err := client.Transactions(nil, &sdk.TransactionsFilter{
		Function: &function,
	})
	if !isNilError(err) {
		log.Fatalf("Failed to get transactions: %v", err)
	}

	for _, transaction := range transactions.Data {
		fmt.Println("Digest:", transaction.Transaction.Digest().ToBase58())
	}
}

File: bindings/wasm/examples/transactions_with_function.mjs
// Copyright (c) 2026 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { GraphQlClient, TransactionsFilter, initAsync } from "@iota/sdk-wasm";

await initAsync();

const client = GraphQlClient.newTestnet();
const transactions = await client.transactions(
  TransactionsFilter.new({ function: "0x3::iota_system::request_add_stake" }),
);
for (const transaction of transactions.data) {
  console.log("Digest:", transaction.transaction.digest().toBase58());
}
